### PR TITLE
Use a constant for JvmOverloads to avoid a crash due to relocation

### PR DESCRIPTION
**Pull request by BoD**
A fix for #4004.

There were also a few other instances or classes used directly, which didn't seem to cause trouble, but also fixed here for consistency.

### CODE REVIEW

**Review by martinbonnin — APPROVED** [review]
👍  I think the JDK classes (like `java.lang.Override` and `java.lang.String`) are not relocated because they're not part of the fat jar so that'd explain why this has worked so far.

Doesn't hurt to not use them reflectively though 👍

**Comment by martinbonnin** [comment]
CI fails with

```
fullstack/LaunchDetailsQuery.kt: (22, 19):  Declaration annotated with '@OptionalExpectation' can only be used in common module sources
```

**Comment by BoD** [comment]
Nothing's ever easy is it? 😅
I'm guessing this is due to the specific wiring in `integration-tests` to have both Java and Kotlin codegens?

I moved the test to a specific submodule instead - JVM only but also tried locally on a MP one which also didn't have the issue.

**Comment by martinbonnin** [comment]
> I'm guessing this is due to the specific wiring in integration-tests to have both Java and Kotlin codegens?

How would that be an issue? Java codegen is not impacted and Kotlin should compile in all cases?

**Comment by BoD** [comment]
> How would that be an issue? Java codegen is not impacted and Kotlin should compile in all cases?

That was kind of a wild guess, but in fact I just noticed it's `compileTestKotlinJs` that's failing 🤔. Will investigate more.

**Comment by BoD** [comment]
I can repro `Declaration annotated with '@OptionalExpectation' can only be used in common module sources` by setting this on my test project:

```
  outputDirConnection {
    connectToKotlinSourceSet("jsTest")
    connectToKotlinSourceSet("jvmTest")
    connectToKotlinSourceSet("appleTest")
  }
```
 
but to be honest I'm still bewildered

File: apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/Options.kt [hunk 1]
```diff
@@ -177,12 +176,13 @@ class Options(
     val generateOptionalOperationVariables: Boolean = defaultGenerateOptionalOperationVariables,
 
     /**
-     * Whether to generate kotlin constructors with @JvmOverloads for more graceful Java interop experience when default values are present.
+     * Whether to generate kotlin constructors with `@JvmOverloads` for more graceful Java interop experience when default values are present.
+     * Note: when enabled in a multi-platform setup, the generated code can only be used in the common or JVM sourcesets.
```
Comment (martinbonnin) [inline apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/Options.kt:180]: 👍

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**